packages/zosuss/src/api/Shell.ts

```
@@ -0,0 +1,73 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Logger } from "@brightside/imperative";
+import { Client, ClientChannel } from "ssh2";
+import { SshSession } from "../SshSession";
+
+export class Shell {
+    public static executeSsh(session: SshSession, command: string, callback: any): void {
+        const conn = new Client();
+        let privateKeyPath;
+        try {
+            privateKeyPath = require("fs").readFileSync(session.ISshSession.privateKey);
+        } catch (err) {
+            privateKeyPath = "";
+        }
+
+        conn.on("ready", () => {
+            conn.shell((err: any, stream: ClientChannel) => {
+                if (err) { throw err; }
+
+                stream.on("close", () => {
+                    conn.end();
+                    process.stdout.write("Closed connection\n");
+                });
+                // exit multiple times in case of multiple shells running in depth
+                stream.end(command + "\nexit\nexit\nexit\nexit\nexit\nexit\nexit\nexit\n");
+                callback(stream);
+            });
+        });
+        conn.connect({
+            host: session.ISshSession.hostname,
+            port: session.ISshSession.port,
+            username: session.ISshSession.user,
+            password: session.ISshSession.password,
+            privateKey: privateKeyPath,
+            passphrase: session.ISshSession.keyPassword
+        });
+        conn.on("error", (err) => {
+            process.stderr.write(err +
```

I think we need a better error message here... at least 'Failed to connect to SSH server'.  Also I don't think this is the right way to pass the error back, I think you should throw an ImperativeError with the message.  Probably shouldn't output the users password directly to the console either.

---

When the error is not handled or when it throws Imperative error, it hangs the terminal for several minutes until the timeout that is set in ssh2 module. I'm trying to find a way to avoid it as well as using imperative error message, but no luck yet.